packages/block-library/src/pullquote/deprecated.js

```diff
@@ -57,7 +58,128 @@ function parseBorderColor( styleString ) {
 	}
 }
 
+// TODO: this is ripe for a bit of a clean up according to the example in https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/#example
 const deprecated = [
+	{
+		attributes: {
+			...blockAttributes,
+		},
+		save( { attributes } ) {
+			const {
+				mainColor,
+				customMainColor,
+				customTextColor,
+				textColor,
+				value,
+				citation,
+				className,
+			} = attributes;
+
+			const isSolidColorStyle = includes( className, SOLID_COLOR_CLASS );
+
+			let figureClasses, figureStyles;
+
+			// Is solid color style
+			if ( isSolidColorStyle ) {
+				const backgroundClass = getColorClassName(
+					'background-color',
+					mainColor
+				);
+
+				figureClasses = classnames( {
+					'has-background': backgroundClass || customMainColor,
+					[ backgroundClass ]: backgroundClass,
+				} );
+
+				figureStyles = {
+					backgroundColor: backgroundClass
+						? undefined
+						: customMainColor,
+				};
+				// Is normal style and a custom color is being used ( we can set a style directly with its value)
+			} else if ( customMainColor ) {
+				figureStyles = {
+					borderColor: customMainColor,
+				};
+			}
+
+			const blockquoteTextColorClass = getColorClassName(
+				'color',
+				textColor
+			);
+			const blockquoteClasses =
+				( textColor || customTextColor ) &&
+				classnames( 'has-text-color', {
+					[ blockquoteTextColorClass ]: blockquoteTextColorClass,
+				} );
```

Comment:
It seems that:

1. If either `textColor` or `customTextColor` is set, we want to set the `has-text-color` className. That figures 👍 
2. If one of those is set, _and_ `blockquoteTextColorClass` is set, we want to set the `[ blockquoteTextColorClass ]` class name. But IIUC, we only get a `blockquoteTextColorClass` if we have a non-custom text color (see [implementation](https://github.com/WordPress/gutenberg/blob/fe379134644e6a8b43a220d1d6e956def9b5295e/packages/block-editor/src/components/colors/utils.js#L49-L64)), so the `customTextColor` criterion is somewhat deceptive.

So can we maybe change this to

```suggestion
			const blockquoteClasses = classnames( {
				'has-text-color': textColor || customTextColor,
				[ blockquoteTextColorClass ]: blockquoteTextColorClass,
			} );
```

for better legibility, and to better match [`figureClasses`](https://github.com/WordPress/gutenberg/pull/30951/files#diff-9f8162c8a4e64eafe65af6866d3d2ffb30cc8dc72b71e5a13ab9f783865b038eR89-R92)?

Reply:
The above is much better, thanks.

I left everything in the save function "as is", since I wasn't sure whether to touch the deprecated save function, or to preserve as it is for succeeding archeological expeditions 😄